src/main/java/com/boomi/flow/services/boomi/mdh/common/Entities.java

```diff
@@ -52,22 +52,28 @@ public static MObject createQuarantineMObject(String universeId, QuarantineEntry
         return new MObject(universeId + "-quarantine", entry.getTransactionId(), properties);
     }
 
-    public static MObject setRandomUniqueIdIfEmpty(MObject object, String idField) {
-        if (Strings.isNullOrEmpty(object.getExternalId()) == false) {
-            return object;
-        }
+    public static MObject setRandomUniqueIdIfEmpty(MObject object, String idField, boolean model) {
         // We are requesting an object without id
         String id = UUID.randomUUID().toString();
 
         // Set the ID property, so it can be referenced in a Flow
         for (Property property : object.getProperties()) {
-            if (property.getDeveloperName().equals(idField)) {
+            if (property.getDeveloperName().equals(idField) && Strings.isNullOrEmpty(object.getExternalId()) == true && model) {
```

Comment:
I don't think you really need this `== true`, but what is `model`? That's not really a boolean word